drivers/adc/adc_stm32.c

```diff
@@ -968,6 +1048,34 @@ static const struct adc_driver_api api_stm32_driver_api = {
 #endif
 };
 
+#define STM32_ADC_INIT_SHARED(index)					\
+									\
+static const struct soc_gpio_pinctrl adc_pins_##index[] =		\
+	ST_STM32_DT_INST_PINCTRL(index, 0);				\
+									\
+static const struct adc_stm32_cfg adc_stm32_cfg_##index = {		\
+	.base = (ADC_TypeDef *)DT_INST_REG_ADDR(index),			\
+	.prescalar = DT_INST_PROP_OR(index, prescalar, 4),		\
+	.irq_cfg_func = adc_stm32_shared_irq_handler,			\
```

Comment:
This part of the struct is actually meant to hold the irq init function, not the irq handler.
In current case, it should then be NULL if we don't want to use it.
Alternatively, it could be set to `adc_stm32_cfg_func_isr` and use the `init_irq` flag inside this function to make sure irq init is only done once (which would remove need for the `adc_stm32_init` hack).